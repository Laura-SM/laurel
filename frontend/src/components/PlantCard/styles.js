import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    flex: 1,
  },
  image: {
    width: 350,
    height: 350,
    borderRadius: 10,
    margin: 20,
  },
  textContainer: {
    position: 'absolute',
    bottom: 20,
    backgroundColor: '#7a918d',
    opacity: 0.6,
    marginHorizontal: 20,
    width: 250,
    height: 55,
    borderTopRightRadius: 60,
    borderBottomLeftRadius: 10,
  },
  text: {
    fontSize: 22,
    marginHorizontal: 40,
    color: '#faf9f9',
    position: 'absolute',
    bottom: 35,
  },
});

export default styles;
