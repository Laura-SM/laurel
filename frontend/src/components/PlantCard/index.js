import React from 'react';
import {Image, Pressable, Text, View} from 'react-native';
import styles from './styles';

const PlantCard = ({plant, navigation}) => {
  return (
    <Pressable
      testID="goToDetails"
      style={styles.cardContainer}
      onPress={() => navigation.navigate('PlantDetails', {plantId: plant._id})}>
      <Image style={styles.image} source={{uri: plant.image}} />

      <View style={styles.textContainer} />
      <Text style={styles.text}>{plant.name}</Text>
    </Pressable>
  );
};

export default PlantCard;
