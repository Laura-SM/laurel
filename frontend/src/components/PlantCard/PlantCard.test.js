import 'react-native';
import React from 'react';
import PlantCard from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a PlantCard component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;

  beforeEach(() => {
    myStore = mockStore({
      name: 'kentia',
      _id: '1a',
      image: 'img.com',
      room: 'bedroom',
    });
    navigation = {
      navigate: jest.fn(),
    };
  });

  describe('When invoques PlantCard function', () => {
    test('Then PlantCard component renders correctly', () => {
      const wrapper = render(
        <Provider store={myStore}>
          <PlantCard plant={myStore} />
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When a plant card is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <PlantCard plant={myStore} navigation={navigation} />
        </Provider>,
      );
      const gotToDetails = getByTestId('goToDetails');
      fireEvent.press(gotToDetails);

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
