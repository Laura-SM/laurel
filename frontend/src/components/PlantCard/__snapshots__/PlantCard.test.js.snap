// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PlantCard component When invoques PlantCard function Then PlantCard component renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
  testID="goToDetails"
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 350,
        "margin": 20,
        "width": 350,
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "#93B1A7",
        "borderBottomLeftRadius": 10,
        "borderTopRightRadius": 60,
        "bottom": 20,
        "height": 55,
        "marginHorizontal": 20,
        "opacity": 0.6,
        "position": "absolute",
        "width": 250,
      }
    }
  />
  <Text
    style={
      Object {
        "bottom": 35,
        "color": "#faf9f9",
        "fontSize": 22,
        "marginHorizontal": 40,
        "position": "absolute",
      }
    }
  />
</View>
`;
